packages/compass-query-bar/src/plugin.jsx

```diff
@@ -1,38 +1,66 @@
-import React, { Component } from 'react';
+import React, { useCallback } from 'react';
 import PropTypes from 'prop-types';
 import { StoreConnector } from 'hadron-react-components';
+import { isFunction } from 'lodash';
+
 import LegacyQueryBar from './components/legacy-query-bar';
 import { QueryBar } from './components/query-bar/query-bar';
 
-class Plugin extends Component {
-  static displayName = 'QueryBarPlugin';
-  static propTypes = {
-    store: PropTypes.object.isRequired,
-    actions: PropTypes.object.isRequired,
-  };
-
-  /**
-   * Connect the Plugin to the store and render.
-   *
-   * @returns {React.Component} The rendered component.
-   */
-  render() {
-    const useNewQueryBar = process?.env?.COMPASS_SHOW_NEW_TOOLBARS === 'true';
-
-    return (
-      <StoreConnector store={this.props.store}>
-        {useNewQueryBar ? (
-          <QueryBar
-            toggleExpandQueryOptions={this.props.actions.toggleQueryOptions}
-            toggleQueryHistory={this.props.actions.toggleQueryHistory}
-          />
-        ) : (
-          <LegacyQueryBar actions={this.props.actions} {...this.props} />
-        )}
-      </StoreConnector>
-    );
-  }
+function Plugin({
+  actions,
+  onApply: _onApply,
+  onReset: _onReset,
+  store,
+  ...props
+}) {
+  const useNewQueryBar = process?.env?.COMPASS_SHOW_NEW_TOOLBARS === 'true';
+
+  const onApply = useCallback(() => {
+    actions.apply();
+
+    if (isFunction(_onApply)) {
+      _onApply();
+    }
+  }, [_onApply, actions]);
+
+  const onReset = useCallback(() => {
+    actions.reset();
+
+    if (isFunction(_onReset)) {
+      _onReset();
+    }
+  }, [_onReset, actions]);
```

Comment:
Moved this function and `onApply` to here from https://github.com/mongodb-js/compass/pull/3128/files#diff-c590b2e9bf139a99b4be59d7e761c1ad61bbe4080dfff45af0dec1511195725eL159 so that the components don't need the higher level prop logic.